include/pybind11/detail/class.h

```diff
@@ -317,9 +317,13 @@ extern "C" inline int pybind11_object_init(PyObject *self, PyObject *, PyObject
 inline void add_patient(PyObject *nurse, PyObject *patient) {
     auto &internals = get_internals();
     auto instance = reinterpret_cast<detail::instance *>(nurse);
+    auto &current_patients = internals.patients[nurse];
     instance->has_patients = true;
+    for (auto &p : current_patients)
```

Comment:
This could be `if (std::find(current_patients.begin(), current_patients.end(), patient) != current_patients.end())`, I suppose. Normally that would be cleaner, but now it's actually lengthier. So probably not worth it.